scripts/receipts.sh

```diff
@@ -104,4 +110,71 @@ function receipts::generate() {
   syft packages "${image}" --output cyclonedx-json --file "${output}"
 }
 
+# Generates syft receipts for each architecture for given oci archives
+function receipts::generate::multi::arch() {
+  local buildArchive runArchive registryPort registryPid localRegistry imageType archiveName imageReceipt
+
+  buildArchive="${1}"
+  runArchive="${2}"
+
+  registryPort=$(get::random::port)
+  localRegistry="127.0.0.1:$registryPort"
+
+  # Start a local in-memory registry so we can work with the oci archives
+  PORT=$registryPort crane registry serve --insecure > /dev/null 2>&1 &
+  registryPid=$!
+
+  # Wait for the registry to be available
+  until crane catalog $localRegistry > /dev/null 2>&1; do
+    sleep 1
+  done
+
+  # Push the oci archives to the local registry
+  jam publish-stack \
+    --build-ref "$localRegistry/build" \
+    --build-archive $buildArchive \
+    --run-ref "$localRegistry/run" \
+    --run-archive $runArchive
+  
+  # Ensure we can write to the BUILD_DIR
+  if [ $(stat -c %u build) = "0" ]; then
+    sudo chown -R "$(id -u):$(id -g)" "$BUILD_DIR"
+  fi
+
+  for imageType in build run; do
+    archiveName="${imageType}.oci"
+    util::print::title "Generating package SBOM for ${archiveName}"
+
+    for imageArch in $(crane manifest "$localRegistry/$imageType" | jq -r '.manifests[].platform.architecture'); do
+      util::print::info "Generating CycloneDX package SBOM using syft for $archiveName on platform linux/$imageArch"
+
+      imageReceipt="${BUILD_DIR}/${imageType}-${imageArch}-${receiptFilename}"
+
+      # Generate the architecture-specific SBOM from image in the local registry
+      syft packages "registry:$localRegistry/$imageType" \
+        --output cyclonedx-json \
+        --file "$imageReceipt" \
+        --platform "linux/$imageArch"
+
+      # Copy the amd64 image receipt as the primary one generated by this script
+      if [ $(util::tools::arch) = "amd64" ]; then
+        cp "$imageReceipt" "${BUILD_DIR}/${imageType}-${receiptFilename}"
+      fi
+    done
+  done
+
+  kill $registryPid
```

Comment:
I think we may want to improve the error handling here - using `trap` or something similar. We need to make sure to clean up the registry if the script errors or is killed by the user.

However, I don't want to block this PR on that, so we can carve out a separate issue to look into it.

Reply:
All set. I had never used `trap` before so thanks for suggesting it.